lib/beaker/dsl/install_utils/puppet_utils.rb

```diff
@@ -584,25 +589,22 @@ def install_puppetlabs_dev_repo ( host, package_name, build_version,
                           repo_filename,
                           platform_configs_dir)
 
-            link = "%s/%s/%s/repos/%s/%s%s/PC1/%s/" %
-              [ dev_builds_url, package_name, build_version, variant,
-                fedora_prefix, version, arch ]
-
-            if not link_exists?( link )
-              link = "%s/%s/%s/repos/%s/%s%s/products/%s/" %
-                [ dev_builds_url, package_name, build_version, variant,
-                  fedora_prefix, version, arch ]
-            end
-
-            if not link_exists?( link )
-              link = "%s/%s/%s/repos/%s/%s%s/devel/%s/" %
+            link = nil
+            package_repos = opts[:dev_builds_repos].nil? ? [] : [opts[:dev_builds_repos]]
+            package_repos.push(['products', 'devel']).flatten!
```

Comment:
@justinstoller @haus is right about how it should be working.

I have questions for you on this, I'll add them to the ticket momentarily